### DISCUSSION TITLE
How does Sentry know the git commit hash when an exception occurs?

### DISCUSSION THREAD

**Discussion by ViggieM** [getsentry/support#19] [Support] [answered]
I am trying to implement a [version endpoint](https://www.youtube.com/watch?v=HMELIR2msdg&list=WL&index=3&t=914s) and I was surprised that Sentry already knows the git commit hash in the "Releases" tab of a project. How does that work??

**Comment by markstory** [comment, marked as answer]
Depending on how your project is setup, Sentry can get git commits from a repository integration with github/gitlab etc, or from deployment automation using `sentry-cli`.

**Reply by ViggieM** [reply]
What seemed weird to me is that I didn't specify a git repository for the project. The only thing I did was integrate sentry in my django project with this guide: https://docs.sentry.io/platforms/python/guides/django/ so at that point I didn't think it would know anything about my git repo. It seems like magic to me :open_mouth:

**Reply by markstory** [reply]
If you configured the `release` attribute in your SDK configuration, sentry will be able to figure out what 'release' issues first/last appeared in. That can be done without a repository integration as well.
